pam-OSC-Module.js

```diff
@@ -12,214 +12,363 @@
 // GNU General Public License for more details.
 
 // You should have received a copy of the GNU General Public License
-// along with this program.  If not, see <https://www.gnu.org/licenses/>. 
-
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 // Todo: Refactor: This is only a temp solution
 var displayDevice = null;
-var colors = ["0;0;0;0", "0;0;0;0", "0;0;0;0", "0;0;0;0", "0;0;0;0", "0;0;0;0", "0;0;0;0", "0;0;0;0"];
-
-const utils = require('./utils.js');
-const colorUtils = require('./colorUtils.js');
-const routingUtils = require('./routingUtils.js');
-const midiUtils = require('./midiUtils.js');
-const oscUtils = require('./oscUtils.js');
+var colors = [
+  "0;0;0;0",
+  "0;0;0;0",
+  "0;0;0;0",
+  "0;0;0;0",
+  "0;0;0;0",
+  "0;0;0;0",
+  "0;0;0;0",
+  "0;0;0;0"
+];
+
+const utils = require("./utils.js");
+const colorUtils = require("./colorUtils.js");
+const routingUtils = require("./routingUtils.js");
+const midiUtils = require("./midiUtils.js");
+const oscUtils = require("./oscUtils.js");
 
 var routing = {};
 
 let encoderFine = false;
 let encoderRough = false;
 let currentAttribute = "dimmer";
+let timecode = {
+  selectedSlot: 0,
+  slots: {}
+};
 
 var prefix = "";
 var page = "1";
 
-const ipPort = ('' + settings.read("send")).split(":");
+const ipPort = ("" + settings.read("send")).split(":");
 const ip = ipPort[0];
 const oscPort = ipPort[1];
 
 settings.read("midi").forEach(deviceMidi => {
-	const name = deviceMidi.split(":")[0];
-	const fileName = name + ".json";
-
-	const value = loadJSON('mappings/' + fileName, (e) => console.error("The Mapping " + fileName + " could not be found. Please make sure it exists, or rename your MIDI Device name to a existing one"));
-	value.buttonFeedbackMapper = eval('(' + value.buttonFeedbackMapper + ')');
-	routing[name] = value;
-	console.log("loaded mapping: ", fileName);
+  const name = deviceMidi.split(":")[0];
+  const fileName = name + ".json";
+
+  const value = loadJSON("mappings/" + fileName, e =>
+    console.error(
+      "The Mapping " +
+        fileName +
+        " could not be found. Please make sure it exists, or rename your MIDI Device name to a existing one"
+    )
+  );
+  value.buttonFeedbackMapper = eval("(" + value.buttonFeedbackMapper + ")");
+  routing[name] = value;
+  console.log("loaded mapping: ", fileName);
 });
 
 midiUtils.sendAttributeLED(routing, currentAttribute);
 
+for (let name of Object.keys(routing)) {
+  midiUtils.resetSegments(routing, name);
```

Comment:
could we add a Check here, if the Device Supports the Timecode slot ?
So that we add a extra Field in the Mapping: `enableTimecodeSend ` and only if the device has this enabled we reset it, and later on we send it.

Reply:
Yup, I can do that. I'm wondering if it may be better though to check if the `mode` field in the current mapping is set to `mc` as that's the assumption made when the segment displays are being set.

I do see a good argument though where if we did add `enableTimecodeSend`, it would be easier down the line to add support for changing the segment displays of non mc mode controllers.